src/classes/UTIL_DMLService.cls

```diff
@@ -0,0 +1,256 @@
+    * @return List<Database.SaveResult>
+    **********************************************************************************************************/
+    public static List<Database.SaveResult> updateRecords(List<sObject> recordsToUpdate, Boolean allOrNone) {
+        Database.DMLOptions dmlOptions = new Database.DMLOptions();
+        
+        dmlOptions.optAllOrNone = allOrNone;
+        dmlOptions.DuplicateRuleHeader.AllowSave = retrieveRespectDuplicateRuleSettings();
+
+        return Database.update(recordsToUpdate, dmlOptions);
+    }
+
+
+    /*********************************************************************************************************
+    * @description Wraps the Database class update method and sets the DMLOptions optAllOrNone accordingly 
+    *   and defaults DuplicateRuleHeader.AllowSave to true.
+    * @param recordToUpdate The sObject to update in the org.
+    * @param dmlOptions The optional parameter that specifies additional data for the transaction.
+    * @return Database.SaveResult
+    **********************************************************************************************************/
+    public static Database.SaveResult updateRecord(sObject recordToUpdate, Database.DMLOptions dmlOptions) {
+        if (!useAllowSaveProperty(dmlOptions)) {
+            dmlOptions.DuplicateRuleHeader.AllowSave = retrieveRespectDuplicateRuleSettings();
+        }
+
+        return Database.update(recordToUpdate, dmlOptions);
+    }
+
+
+    /*********************************************************************************************************
+    * @description Wraps the Database class update method and sets the DMLOptions optAllOrNone accordingly 
+    *   and defaults DuplicateRuleHeader.AllowSave to true.
+    * @param recordsToUpdate The sObjects to update in the org.
+    * @param dmlOptions The optional parameter that specifies additional data for the transaction.
+    * @return List<Database.SaveResult>
+    **********************************************************************************************************/
+    public static List<Database.SaveResult> updateRecords(List<sObject> recordsToUpdate, Database.DMLOptions dmlOptions) {
+        if (!useAllowSaveProperty(dmlOptions)) {
+            dmlOptions.DuplicateRuleHeader.AllowSave = retrieveRespectDuplicateRuleSettings();
+        }
+
+        return Database.update(recordsToUpdate, dmlOptions);
+    }
+
+
+    /*********************************************************************************************************
+    * @description Checks if the Allow Save property is set and uses it accordingly.
+    * @return Boolean
+    **********************************************************************************************************/
+    private static Boolean useAllowSaveProperty(Database.DMLOptions dmlOptions) {
+        return (dmlOptions.DuplicateRuleHeader.AllowSave != null);
```

Comment:
good solution!  respect true or false, and only overwrite with the setting if null!

Reply:
Thanks!